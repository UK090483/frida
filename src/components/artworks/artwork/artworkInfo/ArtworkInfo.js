import React from "react"
import styled from "styled-components"
import PropTypes from "prop-types"

import { ProductName } from "../../../lib/ProductComponents"
const ArtworkInfo = ({ availability, artworkName, price }) => {
  return (
    <Root>
      <NamePriceWrap>
        <ProductName
          size={"s"}
          name={artworkName}
          availability={availability === "availabil"}
        />
        <Price>{price}â‚¬</Price>
      </NamePriceWrap>
    </Root>
  )
}

const Root = styled.div`
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  font-size: 1.35rem;

  font-weight: 500;
`

const NamePriceWrap = styled.div`
  width: 100%;
  display: flex;
  flex-wrap: wrap;
`

const Price = styled.div`
  padding-left: 10px;
  padding-top: 15px;
  margin-left: auto;
  width: fit-content;
  text-align: right;
`

ArtworkInfo.defaultProps = {
  availability: "availabil",
  artworkName: "prop artworkName is missing",
  price: 0,
  theme: {
    colors: {
      green: "green",
      red: "red",
    },
  },
}

ArtworkInfo.propTypes = {
  availability: PropTypes.string,
  artworkName: PropTypes.string,
  price: PropTypes.number,
}
export default ArtworkInfo
