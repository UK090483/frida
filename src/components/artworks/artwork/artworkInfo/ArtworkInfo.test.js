import React from "react"
import renderer from "react-test-renderer"
import ArtworkInfo from "./ArtworkInfo"
import { ThemeProvider } from "styled-components"
const theme = {
  colors: {
    green: "green",
    red: "red",
  },
}
describe("Artwork Info", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={theme}>
          <ArtworkInfo />
        </ThemeProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
