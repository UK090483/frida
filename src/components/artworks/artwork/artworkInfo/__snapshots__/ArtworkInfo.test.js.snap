// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artwork Info renders correctly 1`] = `
<div
  className="sc-dlnjPT ceOMDB"
>
  <div
    className="sc-hKFyIo gAJMfD"
  >
    <div
      className="sc-bdnylx hRWysm"
    >
      <span
        className="sc-gtssRu PXoN"
      >
        prop 
      </span>
      artworkName is missing
    </div>
    <div
      className="sc-eCApGN hKxwXc"
    >
      0
      â‚¬
    </div>
  </div>
</div>
`;
